src/Driver/Element/Wpphp/DatabaseElement.php

```diff
@@ -121,6 +121,9 @@ public function update($id, $args = [])
                 )
             );
         }
+
+        // clear the cache after restoration
+        \wp_cache_flush();
```

Comment:
Why?

If there's a good reason, why don't we need to change the WP-CLI driver to do the same?

Reply:
I put a description in the commit. What was happening is that the first two tests were passing in the feature file, but not the third. The reason was that after the database was restored, wordpress still had the cached user - so believed it existed and wouldn't create it - but also wouldn't let the user log in, because they didn't exist in the database. If you check out https://github.com/paulgibbs/behat-wordpress-extension/pull/150/commits/d18b6eefdac72d4f7c433ee2a1817709a6fc2ecd and run it with WPPHP you'll see that it fails if you have DB restore enabled. I put a similar explanation in the commit message.

I'm not sure why the same thing didn't fail for WP-CLI. As I was writing I assumed that WP-CLI flushed the cache of its own accord. That seems to not be the case. There is a `wp cache` command but that doesn't have a flush option. I'll do some more investigation and see if I can come up with a more satisfactory answer/solution.

Reply:
`wp cache flush` does exist: https://developer.wordpress.org/cli/commands/cache/flush/

I wonder if it's more related to some global state. I know the PHPUnit test suite always has a battle with this. If clearing the cache fixes the immediate problem, why not. :)

Reply:
Understood - I'll chuck a link in the comment above it to this comment so we can work out why we did it.